midend/predication.h

```diff
@@ -113,10 +113,6 @@ class Predication final : public Transform {
     // false for statements that are not dependent.
     std::map<cstring, bool> isStatementDependent;
     const IR::Statement* error(const IR::Statement* statement) const {
-        if (inside_action && ifNestingLevel > 0)
```

Comment:
If you want to delete this code you should delete the whole function and its calls; the function does nothing now.
But I think this should be guided by a policy flag: "report errors when a statement cannot be converted".
Then invoke it with the flag set to "no" in your compiler.

Reply:
Finally, you are not even using this pass in your backend, why are you modifying it?

Reply:
Yes. DPDK backend does not use this pass. Wondering if I should still modify to use the policy flag and call it appropriately from p4test backend?

Reply:
As you wish

Reply:
On a closer look at predication pass, I think we can leave it as it is now because the error is  thrown for the statements which prevent the if-else to Mux conversion.
I reverted the changes made to predication pass.